app/internal/agenda_events.py

```diff
@@ -59,11 +59,39 @@ def get_time_delta_string(start: date, end: date) -> str:
 
 
 def filter_dates(
-        events: List[Event], start: Optional[date],
-        end: Optional[date]) -> Iterator[Event]:
+        events: List[Event], start: date,
+        end: date) -> Iterator[Event]:
     """filter events by a time frame."""
 
     yield from (
         event for event in events
-        if start <= event.start.date() <= end
+        if (
```

Comment:
למה לא לצמצם את הפונקציות האלה לאחת בעזרת ארגומנט ברירת מחדל?

Reply:
I don't understand what @sagizaidor meant, but I do think it's actually a good idea to export this long statement to a separate function :)